cargo-dist/src/platform.rs

```diff
@@ -417,3 +436,108 @@ fn supports(
+        self.rosetta2 |= rosetta2;
+    }
+}
+
+/// Combine two min_libc_versions to get a new min that satisfies both
+fn max_of_min_libc_versions(
+    lhs: &Option<LibcVersion>,
+    rhs: &Option<LibcVersion>,
+) -> Option<LibcVersion> {
+    match (*lhs, *rhs) {
+        (None, None) => None,
+        (Some(ver), None) | (None, Some(ver)) => Some(ver),
+        (Some(lhs), Some(rhs)) => Some(lhs.max(rhs)),
+    }
+}
+
+/// Compute the requirements for running the binaries of this release on its host platform
+fn native_runtime_conditions_for_artifact(
+    dist: &DistGraphBuilder,
+    artifact_id: &ArtifactId,
+) -> RuntimeConditions {
+    let manifest = &dist.manifest;
+    let Some(artifact) = manifest.artifacts.get(artifact_id) else {
+        return RuntimeConditions::default();
+    };
+
+    let mut runtime_conditions = RuntimeConditions::default();
+    for asset in &artifact.assets {
+        let asset_conditions = native_runtime_conditions_for_asset(manifest, &asset.id);
+        runtime_conditions.merge(&asset_conditions);
+    }
+
+    runtime_conditions
+}
+
+fn native_runtime_conditions_for_asset(
+    manifest: &DistManifest,
+    asset_id: &Option<AssetId>,
+) -> RuntimeConditions {
+    let Some(asset_id) = asset_id else {
+        return RuntimeConditions::default();
+    };
+    let Some(asset) = &manifest.assets.get(asset_id) else {
+        return RuntimeConditions::default();
+    };
+    let Some(linkage) = &asset.linkage else {
+        return RuntimeConditions::default();
+    };
+    // This one's actually infallible but better safe than sorry...
+    let Some(system) = manifest.systems.get(&asset.system) else {
+        return RuntimeConditions::default();
+    };
+
+    // Get various libc versions
+    let min_glibc_version = native_glibc_version(system, linkage);
+    let min_musl_version = native_musl_version(system, linkage);
+
+    // rosetta2 is never required to run a binary on its *host* platform
+    let rosetta2 = false;
+    RuntimeConditions {
+        min_glibc_version,
+        min_musl_version,
+        rosetta2,
+    }
+}
+
+/// Get the native glibc version this binary links against, to the best of our ability
+fn native_glibc_version(system: &SystemInfo, linkage: &Linkage) -> Option<LibcVersion> {
+    for lib in &linkage.system {
+        // If this links against glibc, then we need to require that
+        // TODO: make some check equivalent to this!
+        if lib.is_glibc {
+            // TODO: misty's branch has this
+            if let Some(system_glibc) = system.glibc_version {
+                // If there's a system libc, assume that's what it was built against
+                return Some(system_glibc);
+            } else {
+                // If the system has no known libc version use Ubuntu 20.04's glibc as a guess
+                return Some(LibcVersion {
+                    major: 2,
+                    series: 11,
```

Comment:
We should make this the same as our current hardcoded one from the shell template. It'd also be nice to make this a method on `LibcVersion` so we can keep track of where we've inlined it, eg `LibcVersion::default_glibc()` or something.

```suggestion
                    series: 31,
```